CONTRIBUTING.md

```diff
@@ -210,9 +199,63 @@ Number Reservation for Documentation Use ([RFC 5398]). In particular:
 Both ranges have 16 total numbers each. The hexadecimal notation makes it more
 obvious where the range starts and stops.
 
-## Code Style
+## Pull Requests
 
-All code should follow
-[Go language style guide](https://github.com/golang/go/wiki/CodeReviewComments)
-and [Effective Go](https://go.dev/doc/effective_go) for writing readable Go
-code.
+To contribute a pull request:
+
+1.  Make a fork of the
+    [openconfig/featureprofiles](https://github.com/openconfig/featureprofiles)
+    repo and make the desired changes, following the
+    [GitHub Quickstart](https://docs.github.com/en/get-started/quickstart)
+    guide.
+
+    *   New contributions should be in the feature/experimental directory.
+
+1.  When opening a pull request, use a descriptive title and detail. See
+    [Pull Request Title](#pull-request-title) below.
+
+    *   Pull requests should be kept small, ideally under 500 lines. Small
+        changes allow detailed discussions of the additions that are being made
+        to the model, whilst also ensuring that course-corrections can be made
+        early in the process. If a PR is growing to more than 500 lines, it may
+        need to be broken into multiple smaller PRs.
+
+1.  The pull request should include both the suggested feature profile textproto
+    additions, as well as any relevant additions to tests. Tests should be
+    written in Go using the [ONDATRA](https://github.com/openconfig/ondatra)
+    framework.
+
+1.  The automated CI running against each pull request will check the pull
+    request for compliance. The author should resolve any issues found by CI.
+
+1.  One or more peers in the community may review the pull request.
+
+1.  A feature profile repository maintainer will be reponsible for a final
+    review and approval. Only a feature repository maintainer can merge a pull
+    request to the main branch.
+
+The aim of this process is maintain the model quality and approach that
+OpenConfig working group has strived for since its inception in 2014. Thank you
+for your contributions!
+
+### Pull Request Title
+
+Pull request title should start with the scope (i.e. section in the test plan).
+
+```
+RT-5.1 singleton configuration and telemetry, no traffic test yet
+```
+
+The description may add more details as desired to benefit the reviewers. The
+preferred format is:
+
+```
+    * (M) yang/keychain/openconfig-keychain.yang
+      - Add a typedef for referencing a keychain from other modules.
+    * (M) yang/isis/*
+      - Fix support for hello-authentication to allow for references to a
+        specific keychain as defined in the keychain model.
+      - Fix support for authentication of *SNP packets, referencing a
+        keychain that can be used to auth these packets.
+      - move IS-IS model to openconfig-inet-types rather than ietf-inet-types.
+```
```

Comment:
Ideas:
1.  should we request the author include a link to test run output?  (perhaps via a github gist?)   
2. If we can see believable results of a test passing, would that allow skipping going through the step of  experimental folder?

Reply:
1. I'd certainly like to see some evidence from the test run, but I'm not sure if gist is a good way to manage test output. For what it's worth, test log is typically very long, and I don't know if it could be truncated silently.
2. One time I suggested exactly that on a PR whose test code looked very clean and believable, and it also had a clear canonical location, but on second thought I think having the experimental gating might still be a good idea. We may need to first align internally on where the canonical feature location is, e.g. for gNOI.